AK/Format.cpp

```diff
@@ -73,28 +73,32 @@ static void write_escaped_literal(StringBuilder& builder, StringView literal)
     }
 }
 
-static size_t parse_number(StringView input)
+static bool parse_number(GenericLexer& lexer, size_t& value)
 {
-    size_t value = 0;
-
-    for (char ch : input) {
-        value *= 10;
-        value += ch - '0';
+    value = 0;
+
+    bool consumed_at_least_one = false;
+    while (!lexer.is_eof()) {
+        if (StringView { "0123456789" }.contains(lexer.peek())) {
```

Comment:
```suggestion
        if (lexer.next_is(is_digit)) {
```
You can find ctype-like functions compatible with the GenericLexer under its definition in https://github.com/SerenityOS/serenity/blob/e8ef2b18580af33a5eb453a354310692da5371ff/AK/GenericLexer.h#L85